File: frontend/src/microservices/attraction/objects/AttractionTicketTypeResponse.ts
// 本文件定义 attraction 模块的 `AttractionTicketTypeResponse`，作为响应数据并提供 JSON 编解码。

import type { AttractionTicketTypeRuleResponse } from './AttractionTicketTypeRuleResponse'
import type { AttractionTicketSessionResponse } from './AttractionTicketSessionResponse'

export type AttractionTicketTypeResponse = {
  ticketTypeId: string
  ticketTypeName: string
  description: string
  priceAmount: string
  priceCurrency: string
  status: string
  availableFromDate: string
  availableToDate: string
  totalQuantity: number
  validWeekdays: string[]
  availableQuantityForRequestedDate: number | null
  isAvailableForRequestedDate: boolean
  rules: AttractionTicketTypeRuleResponse[]
  sessions: AttractionTicketSessionResponse[]
}
export const attractionTicketTypeResponseFromJson = (json: string): AttractionTicketTypeResponse =>
  JSON.parse(json) as AttractionTicketTypeResponse

export const attractionTicketTypeResponseToJson = (value: AttractionTicketTypeResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/attraction/objects/AttractionTicketTypeResponse.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AttractionTicketTypeResponse(
    ticketTypeId: String,
    ticketTypeName: String,
    description: String,
    priceAmount: String,
    priceCurrency: String,
    status: String,
    availableFromDate: String,
    availableToDate: String,
    totalQuantity: Int,
    validWeekdays: List[String],
    availableQuantityForRequestedDate: Option[Int],
    isAvailableForRequestedDate: Boolean,
    rules: List[AttractionTicketTypeRuleResponse],
    sessions: List[AttractionTicketSessionResponse]
)
object AttractionTicketTypeResponse:
  given sourceEncoder: Encoder[AttractionTicketTypeResponse] = deriveEncoder
  given sourceDecoder: Decoder[AttractionTicketTypeResponse] = deriveDecoder
